crmd/utils.c

```diff
@@ -1161,3 +1161,36 @@ void crmd_peer_down(crm_node_t *peer, bool full)
     crm_update_peer_join(__FUNCTION__, peer, crm_join_none);
     crm_update_peer_expected(__FUNCTION__, peer, CRMD_JOINSTATE_DOWN);
 }
+
+gboolean
+crmd_is_rsc_managed(const char *rsc)
+{
+    int offset = 0;
+    static int xpath_max = 1024;
+    gboolean rv = TRUE;
+    xmlNode *xml_search = NULL;
+    char *xpath_string = NULL;
+    const char *managed = NULL;
+
+    xpath_string = calloc(1, xpath_max);
+    offset += snprintf(xpath_string + offset, xpath_max - offset,
+                       "%s", get_object_path("resources"));
+    offset += snprintf(xpath_string + offset, xpath_max - offset,
+                       "//*[@id=\"%s\"]", rsc);
+    offset += snprintf(xpath_string + offset, xpath_max - offset, "//nvpair[");
+    offset += snprintf(xpath_string + offset, xpath_max - offset,
+                       "@name=\"%s\"", XML_RSC_ATTR_MANAGED);
+    offset += snprintf(xpath_string + offset, xpath_max - offset, "]");
+
+    if (fsa_cib_conn->cmds->query(fsa_cib_conn, xpath_string, &xml_search,
+            cib_sync_call | cib_scope_local | cib_xpath) == pcmk_ok) {
+        managed = crm_element_value(xml_search, XML_NVPAIR_ATTR_VALUE);
+        if (managed && (crm_is_true(managed) == FALSE)) {
+            rv = FALSE;
+        }
+    }
+
+    free(xpath_string);
+    free_xml(xml_search);
+    return rv;
+}
```

Comment:
Basic goal here was to enable this one maintenance-scenario. So I guess unmanaged as 
a result of an error condition we can forget here but the other reasons are of course
valid  ways how the maintenance scenario can be triggered. Just the question if we have
to support them here. Otherwise the behaviour would just fall back to what it was before.
I'm actually more worrying about the opposite case where pengine would react inappropriate
due to unmanaged triggered by e.g. an error condition.

What stays as well is the sync-query. Would moving the code to async-query-callback solve your concerns? This shouldn't be a deal for remote_proxy_cb but for here I have to again think of the
implications.

Do you have an alternative suggestion? e.g. signal from pengine via some kind of attribute?